src/LowestCommonAncestor.php

```diff
@@ -2,35 +2,55 @@
 
 namespace Relaxed\LCA;
 
+use Exception;
 use Fhaculty\Graph\Vertex;
 use Fhaculty\Graph\Graph;
 use Graphp\Algorithms\Search\BreadthFirst;
+use Fhaculty\Graph\Exception\UnderflowException;
 
-class LowestCommonAncestor {
+class LowestCommonAncestor
+{
+    public function __construct(Graph $graph)
+    {
+        $this->graph = $graph;
+    }
 
-  function __construct(Graph $graph) {
-    $this->graph = $graph;
-  }
+    /**
+     * @param Vertex $local
+     * @param Vertex $remote
+     * @param int $direction
+     * Traverse in reverse order the graph when searching by default;
+     * @return bool
+     */
+    public function find(Vertex $local, Vertex $remote, $direction = BreadthFirst::DIRECTION_REVERSE)
```

Comment:
@dickolsson @sharprakeshverma I'm not sure that changing the direction will return the same result, it might return other/wrong value, not LCA. It definitely needs tests.

@sharprakeshverma Please test how `find()` method works with different directions (for example `$lca->find($vertices['node_14'], $vertices['node_15'], BreadthFirst::DIRECTION_FORWARD);`). We have 3 direction: `DIRECTION_FORWARD`, `DIRECTION_REVERSE` and `DIRECTION_BOTH`;